src/Http/Response.php

```diff
@@ -1141,6 +1141,22 @@ public function withDownload(string $filename)
         return $this->withHeader('Content-Disposition', 'attachment; filename="' . $filename . '"');
     }
 
+    /**
+     * Create a new instance with Content-Disposition header set.
+     *
+     * @param string|null $filename The name of the file the browser can download as after viewing file inline
+     * @return static
+     */
+    public function withInline(?string $filename = null)
```

Comment:
Can't we just add an `as` option to `Response::withFile()` as markstory had suggested? Not too thrilled about increasing the API surface for an uncommon use case. Mostly people just craft the URL to get the required filename.

Reply:
Also new features, especially API changes need to target next minor. Otherwise it becomes difficult for users to keep of when a feature was added.

Reply:
We can do that. Wasn't sure if there was value to just setting the disposition instead of requiring a stream using `withFile()`.

Reply:
The inline disposition is not about the filename - it is about better hinting to the browser to preview the file.